core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -325,7 +325,13 @@ class CudaInternal {
   template <bool setCudaDevice = true>
   cudaError_t cuda_malloc_wrapper(void** devPtr, size_t size) const {
     if constexpr (setCudaDevice) set_cuda_device();
+#if !defined(KOKKOS_ENABLE_IMPL_CUDA_EMULATE_UNIFIED_MEMORY)
     return cudaMalloc(devPtr, size);
+#else
+    auto ptr = cudaMallocManaged(devPtr, size, cudaMemAttachGlobal);
+    cudaDeviceSynchronize();
```

Comment:
UVM allocations are not immediately accessible on some archs. Without it some of our tests segfault. Note we generally sync UVM allocations elsewhere, but since this is in emulation mode only I think its the simplest way to handle this.

Reply:
Is the fence missing in `CudaSpace::impl_allocate_common()` then?

Reply:
It's there for `CudaUVMSpace::impl_allocate`. `cuda_malloc_wrapper` is only used in `CudaInternal::initialize`.